Aquality.WinAppDriver/src/Aquality.WinAppDriver/Forms/Form.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Aquality.Selenium.Core.Elements;
+using Aquality.WinAppDriver.Applications;
+using Aquality.WinAppDriver.Elements;
+using Aquality.WinAppDriver.Elements.Interfaces;
+using OpenQA.Selenium;
+using System;
+using System.Drawing;
+using Element = Aquality.WinAppDriver.Elements.Element;
+using IElement = Aquality.Selenium.Core.Elements.Interfaces.IElement;
+using ElementFactory = Aquality.WinAppDriver.Elements.ElementFactory;
+
+namespace Aquality.WinAppDriver.Forms
+{
+    /// <summary>
+    /// Defines base class for any form on any application's window.
+    /// </summary>
+    public abstract class Form : Element, IForm
+    {
+        /// <summary>
+        /// Constructor with parameters.
+        /// </summary>
+        /// <param name="locator">Unique locator of the window.</param>
+        /// <param name="name">Name of the window.</param>
+        /// <param name="parentForm">Parent form. If set to null, search context of <see cref="AqualityServices.Application"/> is used.</param>
+        /// <param name="isRootSession">Determines whether the search of the current form would be performed from the <see cref="IWindowsApplication.RootSession"/> or not.
+        /// If is set to false, search is performed from the application session <see cref="IWindowsApplication.Driver"/>.</param>
+        /// <param name="elementState">Element presence state.</param>
+        protected Form(By locator, string name, IForm parentForm = null, bool isRootSession = false, ElementState elementState = ElementState.Displayed)
+            : base(locator, name, ResolveSearchContextSupplier(parentForm), parentForm == null ? isRootSession : parentForm.IsRootSession, elementState)
```

Comment:
parentForm?.IsRootSession

Reply:
it won't return the same result, e.g. when parentForm is null and isRootSession is true I expect to get the "true" here